source/Irrlicht/CBAWMeshWriter.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (C) 2018 Krzysztof "Criss" Szenk
+// This file is part of the "Irrlicht Engine" and "Build A World".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+// and on http://irrlicht.sourceforge.net/forum/viewtopic.php?f=2&t=49672
+
+#include "CBAWMeshWriter.h"
+
+#include "IFileSystem.h"
+#include "IWriteFile.h"
+#include "irrArray.h"
+#include "ITexture.h"
+#include "coreutil.h"
+#include "irrTypes.h"
+
+namespace irr {
+	namespace scene
+	{
+
+		const char * const CBAWMeshWriter::BAW_FILE_HEADER = "IrrlichtBaW BinaryFile\0\0\0\0\0\0\0\0\0";
+
+		CBAWMeshWriter::~CBAWMeshWriter() {}
+
+		CBAWMeshWriter::CBAWMeshWriter()
+		{
+#ifdef _DEBUG
+			setDebugName("CBAWMeshWriter");
+#endif
+		}
+
+		template<>
+		void CBAWMeshWriter::exportAsBlob<ICPUMeshBuffer>(ICPUMeshBuffer* _obj, uint32_t _headerIdx, io::IWriteFile* _file)
+		{
+			uint8_t* const data = (uint8_t*)std::malloc(m_headers[_headerIdx].blobSizeDecompr);
+			uint8_t* nearestFree = data;
```

Comment:
All this stuff is nice, but not very maintanable.. maybe make a "struct MeshBufferBlobV1 : public Blob" which will construct itself from an ICPUMeshBuffer*

And also use the stack instead of heap! 
Making our exporter faster!

Reply:
Nice idea,  thanks